processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,9 +82,55 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
+	clusterName, err := d.getClusterName()
+	if err != nil {
+		return pcommon.NewResource(), "", err
+	}
+	d.rb.SetK8sClusterName(clusterName)
+
 	return d.rb.Emit(), conventions.SchemaURL, nil
 }
 
+func (d *detector) getClusterName() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
```

Comment:
should this use the context.Context from Detect?  Same comment for context.TODOs below.

Reply:
I'm not the most well versed here on why, but experimentally directly using `ctx` is causing `context deadline exceeded`. Is it preferable to use it directly and set a longer timeout than using `context.TODO()`?

Reply:
I would vote for setting a longer timeout directly (the timeout comes from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/96d53a20663a0f593565fb84abbad99db3ce88da/processor/resourcedetectionprocessor/config.go#L38-L39C23 )

Reply:
Done. I set the default timeout to 11 seconds when there's an EKS detector, which is the cutoff between success and failure in my demo environment for getting the cluster name.